packages/expo-sms/ios/ExpoSMSModule.swift

```diff
@@ -0,0 +1,97 @@
+import ExpoModulesCore
+import MessageUI
+import CoreServices
+import MobileCoreServices
+import UniformTypeIdentifiers
+
+struct ExpoSMSContext {
+  let promise: Promise
+  let smsDelegate: SMSDelegate
+}
+
+public class ExpoSMSModule: Module, SMSResultHandler {
+  private var smsContext: ExpoSMSContext?
+  private lazy var utils = appContext?.utilities
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoSMS")
+
+    AsyncFunction("isAvailableAsync") {
+      return MFMessageComposeViewController.canSendText()
+    }
+
+    AsyncFunction("sendSMSAsync") {(addresses: [String], message: String, options: SmsOptions, promise: Promise) in
+      try sendSMSAsync(addresses: addresses, message: message, options: options, promise: promise)
+    }.runOnQueue(.main)
+  }
+
+  private func sendSMSAsync(addresses: [String], message: String, options: SmsOptions, promise: Promise) throws {
+    if !MFMessageComposeViewController.canSendText() {
+      throw SmsUnavailableException()
+    }
+
+    if smsContext != nil {
+      throw SmsPendingException()
+    }
+
+    let smsDelegate = SMSDelegate(handler: self)
+    let context = ExpoSMSContext(promise: promise, smsDelegate: smsDelegate)
+
+    let messageComposeViewController = MFMessageComposeViewController()
+    messageComposeViewController.messageComposeDelegate = context.smsDelegate
+    messageComposeViewController.recipients = addresses
+    messageComposeViewController.body = message
+
+    for attachment in options.attachments {
+      let utiRef = UTTypeCreatePreferredIdentifierForTag(
+        kUTTagClassMIMEType, attachment.mimeType as CFString, nil)
+
+      if utiRef == nil {
+        context.promise.reject("E_SMS_ATTACHMENT",
+                               "Failed to find UTI for mimeType: \(attachment.mimeType)")
+        return
+      }
+
+      guard let url = URL(string: attachment.uri) else {
+        context.promise.reject("E_SMS_ATTACHMENT", "Inavlid file uri: \(attachment.uri)")
```

Comment:
Same as above + fix a typo:
```suggestion
        context.promise.reject("E_SMS_ATTACHMENT", "Invalid file uri: \(attachment.uri)")
```